crates/block-producer/src/mempool/batch_graph.rs

```diff
@@ -167,6 +167,26 @@ impl BatchGraph {
         Ok(batches)
     }
 
+    /// Remove all batches associated with the given transactions (and their descendents).
+    ///
+    /// Transactions not currently associated with a batch are allowed, but have no impact.
+    ///
+    /// # Returns
+    ///
+    /// Returns all removed batches and their transactions.
+    ///
+    /// Unlike [remove_batches](Self::remove_batches), this has no error condition as batches are
+    /// derived internally.
+    pub fn remove_transactions<'a>(
```

Comment:
nit: since we are technically removing batches (and not transactions), should we rename this into something like `remove_batches_with_transactions()`?